### PR TITLE
Safely call create-or-update-comment when dealing with forks

### LINKED ISSUE #853: Cannot comment about Playwright on PRs from forks
## Description

The action that comments about Playwright failures does not work on PRs from forks. This is a [known limitation](https://github.com/peter-evans/create-or-update-comment#action-inputs) and the workaround (using `pull_request_target`) is not acceptable because we want the tests to run in the context of the PR and not `main`.

## Reproduction

See this [failed run](https://github.com/WordPress/openverse/actions/runs/4330608751/jobs/7591849919).

## Solution

It would seem that the only solution is to print the debugging message in the failed Playwright job itself and skip the comment job for PRs from forks.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #853 by @dhruvkb 

## Description

I'm not sure if my solution is entirely the best one. I like that it avoids us needing to copy the body text between two separate actions in the workflow, but is it more complicated to have an additional layer sitting between the call to the upstream workflow? If it was normal code I'd say not (we wrap method calls to external APIs all the time to handle our own particularities) but with GitHub actions, it's so many more layers of redirection.

What do y'all think? Should I change this to just inline the conditional on the playwright action only and duplicate the body text?

## Testing Instructions

See if everything works on this PR: frontend size check, playwright run check.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by krysal — APPROVED** [review]
Looks good!

File: .github/actions/create-or-update-comment/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  issue-number:
+    default: null
+
+runs:
+  using: "composite"
+  steps:
+    - name: Call upstream action
+      # Not fork
+      if: github.repository == 'WordPress/openverse'
+      uses: peter-evans/create-or-update-comment@v2
```
Comment (AetherUnbound) [inline .github/actions/create-or-update-comment/action.yml:22]: Something I'm not sure of - does/would dependabot update versions within composite actions? I'm trying to find an answer now.

Comment (AetherUnbound) [reply .github/actions/create-or-update-comment/action.yml:22]: Update, I found this link: https://stackoverflow.com/a/74997644/3277713

TL;DR is that we have to add explicit dependabot checks for all our composite action directories 😞

Comment (sarayourfriend) [reply .github/actions/create-or-update-comment/action.yml:22]: Good call, easy enough to add.

**Comment by dhruvkb** [comment]
With the changes in `main`, there are only two usages of the `peter-evans/create-or-update-comment` action:

- `playwright-test-failure-comment`
- `emit-docs`

Out of these, the job `emit-docs` does not run on PRs from forks (and from dependabot).

```js
(
  (
    github.event_name == 'push' &&
    github.repository == 'WordPress/openverse'
  ) ||
  (
    github.event_name == 'pull_request' &&
    github.event.pull_request.head.repo.owner.login == 'WordPress' &&
    github.actor != 'dependabot[bot]'
  )
)
```

so the change will be to expand the `if` condition in `playwright-test-failure-comment.steps[2]` where the action is used, greatly simplifying the PR.

**Comment by sarayourfriend** [comment]
Thanks, @dhruv! I will implement that in the next commit.

**Comment by sarayourfriend** [comment]
@dhruvkb I changed the PR as you suggested, but I am wondering if you have any clever alternatives to putting the help body in the separate step? I wanted to avoid duplicating it, but maybe it's better to just copy/paste it...

**Comment by zackkrida** [comment]
@dhruvkb just following up; @sarayourfriend had a question [here](https://github.com/WordPress/openverse/pull/997#issuecomment-1489459887) and I'm not sure if you've seen it.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
I see no harm in having the message logged every time, so we can drop the 'Log help body' step, and just `echo` to `STDOUT` in the 'Build help body' step itself. That can further simplify the code.

**Review by dhruvkb — APPROVED** [review]
